core/src/wallet/tezos/explorers/ExternalTezosLikeBlockchainExplorer.cpp

```diff
@@ -131,19 +131,36 @@ namespace ledger {
         Future<std::shared_ptr<BigInt>>
         ExternalTezosLikeBlockchainExplorer::getGasPrice() {
             const bool parseNumbersAsString = true;
-            const auto gasPriceField = "gas_price";
+            // Since tzindex 12.01, we don't have gas_price field anymore
+            // We have to calculate it instead from gas_limit and fee
+            const auto gasLimitField = "gas_limit";
+            const auto feeField = "fee";
 
             return _http->GET("block/head")
                     .json(parseNumbersAsString).mapPtr<BigInt>(getContext(), [=](const HttpRequest::JsonResult &result) {
                         auto &json = *std::get<1>(result);
 
-                        if (!json.IsObject() || !json.HasMember(gasPriceField) ||
-                            !json[gasPriceField].IsString()) {
-                            throw make_exception(api::ErrorCode::HTTP_ERROR,
-                                                 fmt::format("Failed to get gas_price from network, no (or malformed) field \"{}\" in response", gasPriceField));
+                        if (!json.IsObject()) {
+                            throw make_exception(
+                                api::ErrorCode::HTTP_ERROR, "Failed to compute gas_price from network, block/head is not a JSON object");
+                        }
+                        if (!json.HasMember(gasLimitField) || !json[gasLimitField].IsString()) {
```

Comment:
please, for the sake of retrocompatibility,  consider to check if `gas_price` is in the payload as a fallback solution